mmv1/products/artifactregistry/Repository.yaml

```diff
@@ -388,14 +406,26 @@ properties:
         properties:
           - !ruby/object:Api::Type::Enum
             name: 'publicRepository'
-            exactly_one_of:
-              - remoteRepositoryConfig.0.docker_repository.0.public_repository
             description: |-
               Address of the remote repository.
             immutable: true
+            conflicts:
+              - remoteRepositoryConfig.0.docker_repository.0.custom_repository
             values:
               - :DOCKER_HUB
-            default_value: :DOCKER_HUB
```

Comment:
@ScottSuarez we need to remove this default value and the ones below because either custom_repository or public_repository must be set, not both. It is unfortunate these defaults were added.

From previous conversations it appears this must be a breaking change because we are removing the default_value right? Assuming it is, do you think this field-level breaking change could be included in a minor release? I was reading through the [breaking change docs](https://googlecloudplatform.github.io/magic-modules/develop/breaking-changes/make-a-breaking-change/#in-minor-releases) but am not quite sure if this must be in the major release breaking change branch

Reply:
My lead suggested to add an control field that disables sending of the `publicRepository`. Then we clean it up in the next major release.

Reply:
Oh we still want to support setting `publicRepository` to make it easy for customers to use docker hub. We just don't want to send that value if the customer specifies a `customRepository` uri.

Reply:
Apologies,

The suggestion is as follows:

Leave `publicRepository` as is, with it's default field:
```
publicRepository: <Default Value>
```

Add a new control field that will tell the resource to use `customRepository`:
```
publicRepository: <Default Value>
customRepository: 'Some Value'
useCustomRepository: true
```

Make a bug to clean up this extra field durring the next major release.

Reply:
@ScottSuarez any chance you'd be able to meet over a video chat quick? I'm having quite a bit of difficulty figuring out how to do this looking at [the mmv1 docs](https://googlecloudplatform.github.io/magic-modules/develop/custom-code/#inject-code-before--after-crud-operations-and-import) and [pre_create examples](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/pre_create)

Reply:
You should be able to provide a virtual_field of `use_custom_repository` and `pre_create`/`pre_update` function that 

1. if `use_custom_repository` is present
  a. if `custom_repository` not set throw an error
  b. remove `public_repository` from the api request
2. else 
  a. remove `custom_repository` and or throw an error.
  
You can also set `use_custom_repository` to be required with `custom_repository`

Reply:
If you still need help after checking that out feel free to schedule a call on my calendar

Reply:
@ScottSuarez thanks for the help offline! Does it make sense to you why the docker test is still failing for this commit - https://github.com/GoogleCloudPlatform/magic-modules/pull/10242/commits/55eb2b81fb4ad5eebb0d34bbefd17f43ed0553f9

Even if I remove that if statement and always delete `publicRepository`, the `TestAccArtifactRegistryRepository_artifactRegistryRepositoryRemoteDockerCustomAuthExample` test still fails with that value being set 😞 

I thought maybe `obj` was losing its references to the child maps, but I also tried resetting the maps with no success (you can see this commented out in the commit), which means deleting the nested key from `obj` doesn't do the trick here

Feel free to put a meeting on my cal or I can put another on yours if you want to discuss offline again

Reply:
hmm I can take a look tomorrow morning!

Reply:
Feel free to set some time on my calendar